File: src/main/java/com/rwbymod/gui/GuiHandler.java
package com.rwbymod.gui;

import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.world.World;
import net.minecraftforge.fml.common.network.IGuiHandler;

public class GuiHandler implements IGuiHandler {
	
	public static final int guiScroll = 0;

	public Object getServerGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z) {
		switch(ID) {
		default: return null;
		}
	}

	public Object getClientGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z) {
		switch(ID) {
		case guiScroll: return new GuiScroll();
		default: return null;
		}
	}

}

File: src/main/scala/com/rwbymod/gui/GuiHandler.scala
package com.rwbymod.gui

import net.minecraftforge.fml.common.network.IGuiHandler
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.world.World

object GuiHandler {
  val guiScroll = 0
}

class GuiHandler extends IGuiHandler {
  import GuiHandler._

  def getServerGuiElement(id:Int, player:EntityPlayer, world:World, x:Int, y:Int, z:Int) = id match {
    case _ => null
  }

  def getClientGuiElement(id:Int, player:EntityPlayer, world:World, x:Int, y:Int, z:Int) = id match {
    case i if i == guiScroll => new GuiScroll()
    case _ => null
  }

}